edit/edit.go

```diff
@@ -248,15 +248,40 @@ func InsertAtEnd(stmt []build.Expr, expr build.Expr) []build.Expr {
 
 // FindRuleByName returns the rule in the file that has the given name.
 // If the name is "__pkg__", it returns the global package declaration.
-func FindRuleByName(f *build.File, name string) *build.Rule {
+func FindRuleByName(f *build.File, pkg string, name string) *build.Rule {
 	if name == "__pkg__" {
 		return PackageDeclaration(f)
 	}
 	i := IndexOfRuleByName(f, name)
 	if i != -1 {
 		return &build.Rule{Call: f.Stmt[i].(*build.CallExpr)}
 	}
+	return UseImplicitName(f, pkg, name)
+}
+
+// UseImplicitName returns the rule in the file if it meets these conditions:
+// - It is the only rule in the file.
+// - The passed package and rule names match.
+func UseImplicitName(f *build.File, pkg string, rule string) *build.Rule {
+	ruleCount := 0
+	var r *build.Rule
 
+	for _, stmt := range f.Stmt {
+		call, ok := stmt.(*build.CallExpr)
+		if !ok {
+			continue
+		}
+		r = &build.Rule{Call: call}
+		if r.Kind() != "" {
+			ruleCount++
```

Comment:
Nothing is checking whether the rule has a name here. That means that this code will only work if the BUILD file has exactly one target, rather than when it has exactly one unnamed target. It would be nice to add the name check here, so we don't have that restriction :)